src/unix/linux_like/linux/mod.rs

```diff
@@ -650,6 +650,57 @@ s! {
         pub val: ::c_int,
     }
 
+    // linux/ptp_clock.h
+    pub struct ptp_clock_time {
+        pub sec: ::__s64,
+        pub nsec: ::__u32,
+        pub reserved: ::__u32,
+    }
+
+    pub struct ptp_clock_caps {
+        pub max_adj: ::c_int,
+        pub n_alarm: ::c_int,
+        pub n_ext_ts: ::c_int,
+        pub n_per_out: ::c_int,
+        pub pps: ::c_int,
+        pub n_pins: ::c_int,
+        pub cross_timestamping: ::c_int,
+        #[cfg(target_env = "gnu")]
+        pub adjust_phase: ::c_int,
+        #[cfg(target_env = "gnu")]
+        pub max_phase_adj: ::c_int,
+        #[cfg(target_env = "gnu")]
+        pub rsv: [::c_int; 11],
+        #[cfg(any(target_env = "musl", target_env = "ohos"))]
+        pub rsv: [::c_int; 13],
```

Comment:
Could you add a note about how this is a kernel version difference that happens to be exposed by glibc / musl supported versions?

Also if you have links to relevant sources or docs, please add them in the PR description. I haven't been able to find anything relevant.

Reply:
I'm confused by this now. I suspect that I came to the above after some fighting with CI. Here     // https://github.com/torvalds/linux/blob/b311c1b497e51a628aa89e7cb954481e5f9dced2/include/uapi/linux/ptp_clock.h#L94
 the `max_phase_adj` field clearly is defined. But then e.g. on godbolt, for both musl and gnu, that field does not exist. `adjust_phase` does though. I'll change it back and see what CI says

Reply:
> I'm confused by this now.

At least that makes two of us :)

Reply:
Oh, I just realized this is targeting 0.2 rather than `main`. I would prefer this target `main` first since it uses some newer versions, then for the backport we can work around any kind of legacy weirdness.

Don't close this since it is still useful, but if it isn't too much trouble then it would be good if you could reapply this on top of `main` and open a new PR with that.

(If it's a PITA don't worry about it and I'll figure out how to get things in sync after)

Reply:
worked out fine https://github.com/rust-lang/libc/pull/3865